packages/hebao_v1/contracts/modules/dapps/LRCStakingModule.sol

```diff
@@ -17,15 +17,177 @@
 pragma solidity ^0.5.13;
 pragma experimental ABIEncoderV2;
 
-import "../security/SecurityModule.sol";
+import "../../lib/ERC20.sol";
+import "../../lib/MathUint.sol";
+
+import "../../thirdparty/loopring/IProtocolFeeVault.sol";
+import "../../thirdparty/loopring/IUserStakingPool.sol";
+
+import "./DAppModule.sol";
 
 
 /// @title LRCStakingModule
-contract LRCStakingModule is SecurityModule
+contract LRCStakingModule is DAppModule
 {
-    constructor(Controller _controller)
+    using MathUint for uint;
+
+    IUserStakingPool  public stakingPool;
+    IProtocolFeeVault public feeVault;
+
+    address public lrcTokenAddress;
+
+    constructor(
+        Controller       _controller,
+        IUserStakingPool _stakingPool,
+        address          _lrcTokenAddress
+        )
         public
         SecurityModule(_controller)
     {
+        require(_lrcTokenAddress != address(0), "ZERO_ADDRESS");
+
+        stakingPool = _stakingPool;
+        feeVault = IProtocolFeeVault(stakingPool.protocolFeeVaultAddress());
+        lrcTokenAddress = _lrcTokenAddress;
+    }
+
+    function deposit(
+        address            wallet,
+        address            token,
+        uint               amount
+        )
+        external
+        nonReentrant
+        onlyFromMetaTxOrWalletOwner(wallet)
+        onlyWhenWalletUnlocked(wallet)
+    {
+        require(token == lrcTokenAddress, "LRC_ONLY");
+        require(
+            amount > 0 &&
+            amount <= tokenDepositable(wallet, token),
+            "INVALID_AMOUNT"
+        );
+
+        bytes memory txData = abi.encodeWithSelector(
+            ERC20(0).approve.selector,
+            address(stakingPool),
+            amount
```

Comment:
The safest thing to do here for dapps could be to use
```
uint amountToApprove = amount.add(ERC20(token).allowance(wallet, address(stakingPool));
```
or not setting the approval amount at all depending on the expected behaviour (use up the previously set approved amount or not). Current approvals (like the ones set in `approveToken` in `ApprovedTransfers`) would otherwise be overridden.

But it's unlikely that this would give any issues anyway so it probably doesn't matter.

Reply:
I think it's OK.